src/Microsoft.Extensions.Configuration.UserSecrets/ConfigurationExtensions.cs

```diff
@@ -76,15 +24,15 @@ public static IConfigurationBuilder AddUserSecrets(this IConfigurationBuilder co
                 throw new ArgumentNullException(nameof(configuration));
             }
 
-            var projectPath = configuration.GetProjectPath();
-            if (string.IsNullOrEmpty(projectPath))
+            var fileProvider = configuration.GetFileProvider();
+            if (fileProvider == null)
             {
                 throw new InvalidOperationException(Resources.FormatError_MissingBasePath(
-                        projectPath,
                         typeof(IConfigurationBuilder).Name,
-                        ProjectPathKey));
+                        "FileProvider"));
```

Comment:
I don't think this exception will make sense any more. `GetFileProvider()` will always create a PhysicalFileProvider. If there are any exceptions with the base path it will happen upstream.